docs/development/CODE_REVIEW_GUIDELINES.md

```diff
@@ -0,0 +1,121 @@
+
+For decades, the IETF (Internet Engineering Task Force) has used the motto "rough consensus and running code".
+This motto encapsulates ("running code") that developers' core output is still software: if there is no code that runs and produces correct results, we have nothing.
+It also encapsulates ("rough consensus") that we may not always precisely agree and that's okay.
+
+## Code Reviews
+
+One tactic often employed to produce "maintainable" or "more-readable" code is a "code review".
+These can take many shapes and forms and often have goals beyond simply "code quality".
+
+Broadly speaking, code reviews involve developers looking at some proposed new code (or code changes).
+This is *usually* developers other than the author (although a "self-review" can also be employed).
+In many projects, such attention is a scarce commodity: most programmers would rather write code than read someone else's.
+
+## Code Review Guidelines
+
+All participants must adhere to the overall idea that the review is an attempt to achieve "better" code.
+This is a vague statement on purpose.
+Participants must be cautious in their criticism and generous with praise.
+Participants must remember the scarcity of another developer's attention.
+
+### Expectations
+
+**Pull Request Authors:**
+The author is responsible for tracking up-to-date next actions for a Pull Request to progress towards being merged.
+
+**Reviewers:**
+Reviewers should prefer engaging in code review over starting new work (i.e. taking planned work items that haven't been started yet).
+
+Reviewers (and prospective reviewers) are encouraged to engage in reviews of codebases outside the projects and technologies they use on a day-to-day basis (but not expected to provide an approving review).
+
+## Best Practices
+
+### Reviewing
+
+### Starting a Review
+
+Consider if it could be broken into smaller Pull Requests. If it is clear that it can be, summarize your thinking on how in your Review.
+
+If the commits be (re)organized (i.e. reordered and/or amended) such that there is a commit at which the tests are passing prior to the conclusion of the main change, that's a signal that there's likely a logic split which can be made at that point in such a (re)arrangement.
+
+Use the following guidelines to evaluate whether a Pull Request should be approved:
+
+- *Every* Pull Request should have tests or justification otherwise; esp. bug fixes.
+- *Every* Pull Request should have at least 1 approval from a team member internal or external to the project. Exceptions made by repository maintainer(s), as necessary, on a case-by-case basis.
+
+During review, submit feedback using line comments (A); prefer "Add\[ing a\] single comment" over "Start[ing] a review" (B).
+
+*(Once a review has been started, the option to add single comments is removed. Preferring single comments allows feedback to happen even in the event of an interrupted review.)*
+
+### A
```

Comment:
These A-B-C section headers (which are not in the ToC) are a bit confusing. What is the intention here?

Reply:
The intention was for those headers to act as labels for the respective images which they contain. The A, B, C labels are referenced in the preceding paragraphs like so: "(A)"

Do you have any suggestions for how to better communicate that intention and/or organize things here?

Reply:
The heading can be `Figure A: XXX` or `Figure 1: XXX` and you can reference it accordingly.